lib/screens/v2/candidate_generator/on_bus/departures.ex

```diff
@@ -80,6 +82,47 @@ defmodule Screens.V2.CandidateGenerator.Widgets.OnBus.Departures do
     Enum.filter(departures, &(&1.prediction.route.id != route_id))
   end
 
+  defp filter_duplicate_routes(departures) do
+    unique_departures =
+      Enum.uniq_by(departures, fn dep ->
+        {dep.prediction.route.id, dep.prediction.trip.direction_id}
+      end)
+
+    # Return at least 3 departures if they exist, even if there are not enough unique routes
+    case length(unique_departures) do
+      count when count >= 3 ->
+        unique_departures
+
+      _ ->
+        unique_departures ++
+          Enum.take(departures -- unique_departures, 3 - length(unique_departures))
+    end
+  end
+
+  defp sort_by_mode(departures) do
+    priority = %{
+      :subway => 1,
+      :light_rail => 1,
+      :bus => 2,
+      :commuter_rail => 3,
+      :ferry => 4
+    }
```

Comment:
This could be a module attribute, and I think that would make the function a bit more concise.